pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,464 @@
+package grouppermission
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	operatorconfig "github.com/openshift/rbac-permissions-operator/config"
+	operatorapi "github.com/openshift/rbac-permissions-operator/pkg/apis"
+	managedv1alpha1 "github.com/openshift/rbac-permissions-operator/pkg/apis/managed/v1alpha1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	realclient "sigs.k8s.io/controller-runtime/pkg/client"
+	client "sigs.k8s.io/controller-runtime/pkg/client/fake"
```

Comment:
Another semantic thing, but I think it would be more clear to let this one be named `fake` and let `realclient` be `client`